manim/mobject/text/tex_mobject.py

```diff
@@ -301,42 +311,45 @@ def __init__(
         if self.organize_left_to_right:
             self._organize_submobjects_left_to_right()
 
-    def _break_up_tex_strings(self, tex_strings):
+    def _break_up_tex_strings(self, tex_strings: Sequence[str]) -> list[str]:
         # Separate out anything surrounded in double braces
         pre_split_length = len(tex_strings)
-        tex_strings = [re.split("{{(.*?)}}", str(t)) for t in tex_strings]
-        tex_strings = sum(tex_strings, [])
-        if len(tex_strings) > pre_split_length:
+        # TODO: do we need that str(t)?
+        pre_pieces_arr = [re.split("{{(.*?)}}", str(t)) for t in tex_strings]
+        pre_pieces = sum(pre_pieces_arr, [])
+        if len(pre_pieces) > pre_split_length:
             self.brace_notation_split_occurred = True
 
         # Separate out any strings specified in the isolate
         # or tex_to_color_map lists.
-        patterns = []
-        patterns.extend(
-            [
-                f"({re.escape(ss)})"
-                for ss in it.chain(
-                    self.substrings_to_isolate,
-                    self.tex_to_color_map.keys(),
-                )
-            ],
-        )
+        patterns: list[str] = self.substrings_to_isolate.copy()
+        for key in self.tex_to_color_map:
+            try:
+                # If the given key behaves like tex_strings
+                key + ""  # type: ignore [operator]
+                patterns.append(key)  # type: ignore [arg-type]
+            except TypeError:
+                # If the given key is a tuple
+                patterns.extend(key)
```

Comment:
Wouldn't it make more sense to just use `isinstance`? Something like this
```suggestion
            if isinstance(key, str):
                patterns.append(key)
            else:
                patterns.extend(key)
```

Reply:
It really would. The way in which the comment was worded, `behaves like tex_strings` rather than `is a string`, made me think that maybe the original author thought that `key` could possibly be a non-string object to which you can add a string. However, in the context of tex strings, that doesn't really make sense.